geo-types/src/line_string.rs

```diff
@@ -281,6 +288,61 @@ impl<T: CoordinateType> IndexMut<usize> for LineString<T> {
     }
 }
 
+#[cfg(test)]
+impl<T> RelativeEq for LineString<T>
+where
+    T: CoordinateType + Float + AbsDiffEq,
+{
+    // type Epsilon = T;
+
+    #[inline]
+    fn default_max_relative() -> Self::Epsilon {
+        T::epsilon()
+    }
+
+    #[inline]
+    fn relative_eq(
+        &self,
+        other: &LineString<T>,
+        epsilon: Self::Epsilon,
+        max_relative: Self::Epsilon,
+    ) -> bool {
+        if self.num_coords() != other.num_coords() {
+            return false;
+        }
+
+        let points_zipper = self.points_iter().zip(other.points_iter());
+        for (lhs, rhs) in points_zipper {
+            if lhs.relative_ne(&rhs, epsilon, max_relative) {
+                return false;
+            }
+        }
+
+        true
+    }
+}
+
+#[cfg(test)]
+impl<T: AbsDiffEq + CoordinateType + Float> AbsDiffEq for LineString<T> {
+    type Epsilon = T;
+
+    #[inline]
+    fn default_epsilon() -> Self::Epsilon {
+        T::epsilon()
+    }
+
+    #[inline]
+    fn abs_diff_eq(&self, other: &LineString<T>, epsilon: Self::Epsilon) -> bool {
```

Comment:
Can we do the num_coords check optimization here too like you did in relative_eq?

Reply:
Thanks this was fixed in the commit 

line_string.rs -  abs_diff_eq() Now rejects line_strings with unequal lengths. (with tests)